picard/util/checkupdate.py

```diff
@@ -0,0 +1,159 @@
+        # as defined in PEP-440.  The Picard developers have standardized on using
+        # only 'dev' or 'final' as the str_type segment of the version tuple.  Each
+        # key in the 'urls' dictionary contains a string with the specified url.
+        # Valid keys include: 'download' and 'changelog'.  The only required key in
+        # the 'urls' dictionary is 'download'.
+
+        # Initialize empty dictionary for 'stable' key
+        self._available_versions = {
+            'stable': { 'tag': '', 'version': (0, 0, 0, 'dev', 0), 'urls': {'download': ''} },
+        }
+        self._show_always = False
+        self._update_level = 0
+
+    def check_update(self, show_always=False, update_level=0):
+        '''Checks if an update is available.
+
+        Compares the version number of the currently running instance of Picard
+        and displays a dialog box informing the user  if an update is available,
+        with an option of opening the download site in their browser.  If there
+        is no update available, no dialog will be shown unless the "show_always"
+        parameter has been set to True.  This allows for silent checking during
+        startup if so configured.
+
+        Args:
+            show_always: Boolean value indicating whether the results dialog
+                should be shown even when there is no update available.
+            update_level: Determines what type of updates to check.  Options are:
+                0 = only stable release versions are checked.
+                1 = stable and beta releases are checked.
+                2 = stable, beta and dev releases are checked.
+
+        Returns:
+            none.
+
+        Raises:
+            none.
+        '''
+        self._show_always = show_always
+        self._update_level = update_level
+
+        if self._available_versions['stable']['tag']:
+            # Release information already acquired from specified website api.
+            self._display_results()
+        else:
+            # Gets list of releases from specified website api.
+            self._query_available_updates()
+
+    @property
+    def available_versions(self):
+        '''Provide a list of the latest version tuples for each update type.'''
+        return self._available_versions
+
+    def _query_available_updates(self, callback=None):
+        '''Gets list of releases from specified website api.'''
+        output_text = _("Getting release information from %s." % (VERSIONS_API['host'],))
+        log.debug(output_text)
+        self.tagger.webservice.get(
+            VERSIONS_API['host'],
+            VERSIONS_API['port'],
+            VERSIONS_API['endpoint'],
+            partial(self._releases_json_loaded, callback=callback),
+            parse_response_type=None,
+            priority=True,
+            important=True
+        )
+
+    def _releases_json_loaded(self, response, reply, error, callback=None):
```

Comment:
`callback` is unused here.
I think it should be called, if set, and error/success passed to it, so you can move `config.persist['last_update_check'] = datetime.date.today().toordinal()` out of this (dropping dependency on datetime module), and move it to the caller.

Reply:
Okay, I've made the change as requested.  If I've misunderstood, please let me know.  Thanks.